server/models/sqls/SQLCase1.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Case1, search DB via active_tree_region table,
+ */
+
+class SQLCase1{
+
+  constructor(){
+    this.isFilteringByUserId = false;
+    this.userId = undefined;
+  }
+
+  addFilterByUserId(userId){
+    this.isFilteringByUserId = true;
+    this.userId = userId;
+  }
+
+  getJoin(){
+    let result = "";
+    if(this.isFilteringByUserId){
+      result += "JOIN trees ON tree_region.tree_id = trees.id";
+    }
+    return result;
+  }
+
+  getFilter(){
+    let result = "";
+    if(this.isFilteringByUserId){
+      result += "AND planter_id = " + this.userId + " ";
+    }
+    if(this.treeIds && this.treeIds.length > 0){
+      result += "AND tree_region.tree_id IN(" + this.treeIds.join(",") + ") ";
+    }
+    if(this.mapName){
+      result += `
+        AND tree_region.tree_id IN(
```

Comment:
Still use `in` cause to filter trees of organization, I tried `join`, but seems it's slower than this way.
And I believe we can do better on this to take less time executing this.